model/metrics/dotnet-metrics.yaml

```diff
@@ -0,0 +1,213 @@
+    instrument: updowncounter
+    unit: "{thread}"
+    stability: experimental
+
+  - id: metric.dotnet.thread_pool.work_item.count
+    type: metric
+    metric_name: dotnet.thread_pool.work_item.count
+    brief: >
+      The number of work items that the thread pool has completed since
+      the process has started.
+    note: "This metric reports the same values as calling [`ThreadPool.CompletedWorkItemCount`](https://learn.microsoft.com/dotnet/api/system.threading.threadpool.completedworkitemcount)."
+    instrument: counter
+    unit: "{work_item}"
+    stability: experimental
+
+  - id: metric.dotnet.thread_pool.queue.length
+    type: metric
+    metric_name: dotnet.thread_pool.queue.length
+    brief: >
+      The number of work items that are currently queued to be processed by the
+      thread pool.
+    note: "This metric reports the same values as calling [`ThreadPool.PendingWorkItemCount`](https://learn.microsoft.com/dotnet/api/system.threading.threadpool.pendingworkitemcount)."
+    instrument: updowncounter
+    unit: "{work_item}"
+    stability: experimental
+
+  - id: metric.dotnet.timer.count
+    type: metric
+    metric_name: dotnet.timer.count
+    brief: "The number of timer instances that are currently active."
+    note: "This metric reports the same values as calling [`Timer.ActiveCount`](https://learn.microsoft.com/dotnet/api/system.threading.timer.activecount)."
+    instrument: updowncounter
+    unit: "{timer}"
+    stability: experimental
+
+  - id: metric.dotnet.assembly.count
+    type: metric
+    metric_name: dotnet.assembly.count
+    brief: "The number of .NET assemblies that are currently loaded."
+    note: "This metric reports the same values as calling [`AppDomain.CurrentDomain.GetAssemblies().Length`](https://learn.microsoft.com/dotnet/api/system.appdomain.getassemblies)."
+    instrument: updowncounter
+    unit: "{assembly}"
+    stability: experimental
+
+  - id: metric.dotnet.exceptions.count
+    type: metric
+    metric_name: dotnet.exceptions.count
+    brief: "The number of exceptions that have been thrown in managed code."
+    note: "This metric reports the same values as calling [`AppDomain.CurrentDomain.FirstChanceException`](https://learn.microsoft.com/dotnet/api/system.appdomain.firstchanceexception)."
+    instrument: counter
+    unit: "{exception}"
+    stability: experimental
+    attributes:
+      - ref: error.type
+        requirement_level: required
+
+  - id: metric.dotnet.cpu.count
+    type: metric
+    metric_name: dotnet.cpu.count
+    brief: "The number of processors available to the process."
+    note: "This metric reports the same values as accessing [`Environment.ProcessorCount`](https://learn.microsoft.com/dotnet/api/system.environment.processorcount)."
+    instrument: updowncounter
+    unit: "{cpu}"
+    stability: experimental
+
+  - id: metric.dotnet.cpu.time
+    type: metric
+    metric_name: dotnet.cpu.time
```

Comment:
@braydonk would be interested to hear your and @open-telemetry/semconv-system-approvers thoughts on this in the context of https://github.com/open-telemetry/semantic-conventions/issues/1161

Based on the decisions made in the past (see https://github.com/open-telemetry/semantic-conventions/pull/1035#discussion_r1626774736) we don't report `process.*` metrics from runtime instrumentations.
So we're adding `dotnet.cpu.time` here. 

Still, there is little-to-no difference in how this metric is defined vs `process.cpu.time`.  I assume some languages would end up reporting `process.cpu.time` - see https://github.com/open-telemetry/semantic-conventions/pull/1066#issuecomment-2200489237.

So, we end up having `{runtime}.cpu.*` metrics which are duplicates of process ones or `process.cpu.*` metrics without resource attributes. I think none of those is the desired outcome, so I'm still looking for the guidance on how we could resolve it.

Reply:
I notice that the description of this metric mentions specifically:
```
CPU time used by the process as reported by the CLR.
```
This is different than `process.cpu.time`, which - while not specifically mentioned in that description like it should be - is meant to represent the CPU Time of the process as reported by the operating system/host machine. Someone could theoretically want to instrument both measurements from the OS and measurements from the CLR, and the different namespace for the metric would distinguish them in a much simpler way than looking at the same `process.cpu.time` metric and deciding based on the exact way the metric with the same name was reported differently in each scenario.

So I think I personally land on this being okay as a duplicate metric definition since there is a distinct reporting source, and imo there is room for both definitions to exist because someone could instrument both at the same time. In some runtimes, these values are actually markedly different despite sharing a name, but I don't know enough about CLR to say whether that's true here.

I encourage other System Semconv WG members to weigh in.

Reply:
the runtime gets CPU time from the OS too, so there should be no difference between `dotnet` or `process` metric.
Same seems to be true for java - https://github.com/adoptium/jdk/blob/f7af4504a804711d93208b763b3e41eafcf61735/src/jdk.management/unix/native/libmanagement_ext/OperatingSystemImpl.c#L163 , node.js. 

The problem with duplication is that every runtime would need a different dashboard/alerts/etc because it reports the same data but under different names.

Reply:
> I encourage other System Semconv WG members to weigh in.

If we can ensure that a runtime provides the very same metrics with the OS then I guess it should fine for an instrumented process to report using the `process.*` namespace.

If we still need to distinguish those for reasons, I guess this can be done through an additional attribute like `agent.name: otlp` or `agent.name: otelcol`.

We sth similar in K8s/containers (in implementations) where we emit `container.*` metrics in cases we know that K8s provide metrics which match what the container runtime gives, while we go with `k8s.container.*` ones in case we have sth which is k8s specific.

Reply:
We already have **_stable_** [semantic conventions for JVM](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmcputime) including metrics like `jvm.cpu.time` which sets a precedent. And I'd assume there were good reasons to have an explicit `jvm.cpu.time` vs. using `process.cpu.time` (@trask do you remember?).

With that, I think we should follow that example for consistency reasons and do the same (or something similar) for other languages as well, hence, defining `dotnet.cpu.time` instead of falling back to `process.cpu.time`. Otherwise it raises the question on why it's being done that way in Java but the other way in .NET.

Reply:
Java case is discussed in https://github.com/open-telemetry/semantic-conventions/pull/1035#discussion_r1626774736

Java runtime only returns total time - https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getProcessCpuTime() 
https://eclipse.dev/openj9/docs/api/jdk11/jdk.management/com/ibm/lang/management/OperatingSystemMXBean.html

it does not differentiate between user and system. This is a good reason to report a different metric.

But even if was the case that Java went ahead with java-specific metric, it does not mean than we should not have consistency across other languages if it makes sense.

Reply:
Based on the discussion in the Semconv WG, the conclusion is to proceed with `dotnet.cpu.time`  https://github.com/open-telemetry/semantic-conventions/issues/1215